address-geocoder.php

```diff
@@ -107,7 +107,10 @@ function is_geocoder_needed() {
      */
     function add_meta_boxes( $post_type ) {
         if ( in_array( $post_type, $this->available_post_types ) && 'exclude' != $this->options[ $post_type ] ) {
-            add_meta_box( 'martygeocoder', 'Geocoder', array( $this, 'meta_box_html' ), $post_type, 'normal', 'high' );
+            $current_post_type = get_post_type_object( $post_type );
+            $metabox_title = empty( $this->options['meta-box-title'] ) ? $current_post_type->labels->singular_name . ' Location' : $this->options['meta-box-title'];
```

Comment:
It's worth noting that I am not 100% certain all post types have a singular name should someone not define one when registering a custom post type but it looks like it defaults to Post/Page if not defined explicitly.